aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbTableOperations.java

```diff
@@ -116,20 +118,30 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
                   .build());
       checkMetadataLocation(table, base);
       Map<String, String> properties = prepareProperties(table, newMetadataLocation);
-      persistTable(tableKey, table, properties);
+      persistTable(tableKey, table, properties, retryDetector);
       commitStatus = CommitStatus.SUCCESS;
-    } catch (ConditionalCheckFailedException e) {
-      throw new CommitFailedException(
-          e, "Cannot commit %s: concurrent update detected", tableName());
     } catch (CommitFailedException e) {
       // any explicit commit failures are passed up and out to the retry handler
       throw e;
     } catch (RuntimeException persistFailure) {
-      LOG.error(
-          "Confirming if commit to {} indeed failed to persist, attempting to reconnect and check.",
-          fullTableName,
-          persistFailure);
-      commitStatus = checkCommitStatus(newMetadataLocation, metadata);
+      boolean conditionCheckFailed = persistFailure instanceof ConditionalCheckFailedException;
+
+      // If we got an exception we weren't expecting, or we got a ConditionalCheckFailedException
+      // but retries were performed, attempt to reconcile the actual commit status.
+      if (!conditionCheckFailed || retryDetector.retried()) {
+        LOG.error(
+            "Confirming if commit to {} indeed failed to persist, attempting to reconnect and check.",
+            fullTableName,
+            persistFailure);
```

Comment:
Should this be at a `warn` level instead? If the commit did end up being successful, then it may mislead people if it's at error level. also nit on the message, maybe "Received unexpected failure when committing, validating if commit ended up succeeding"